app/services/foreman/parameters/validator.rb

```diff
@@ -13,6 +13,10 @@ def validate!
           validate_regexp
         when "list", "array"
           validate_list
+        when "hash"
```

Comment:
Right, I see it now, thanks.  That comes back to the additional rescue correctly catching the right exception under 1.9.3.

Now I'm left with this point:

> b) This particular line is actually in the validator class. If you trace where it's called from, it's in LookupKey#validate_default_value and the @options[:type] is actually LookupKey#validator_type. That's only ever going to be none, regexp or list. hash, yaml, json are key_types, not validator_types.

I don't see why these changes are required, or that they make any difference.